classes/Bulk/Bulk.php

```diff
@@ -510,14 +510,42 @@ public function bulk_optimize_callback() {
 	}
 
 	/**
-	 * Launch the missing WebP versions generation
+	 * Launch the missing Next-gen versions generation
 	 *
 	 * @return void
 	 */
 	public function missing_nextgen_callback() {
 		imagify_check_nonce( 'imagify-bulk-optimize' );
 
-		$contexts = explode( '_', sanitize_key( wp_unslash( $_GET['context'] ) ) );
+		$data = $this->generate_next_gen_versions_callback();
+
+		if ( false === $data['success'] ) {
+			wp_send_json_error( [ 'message' => $data['message'] ] );
+		}
+
+		wp_send_json_success( [ 'total' => $data['message'] ] );
+	}
+
+	public function maybe_generate_avif_on_save($args = null) {
```

Comment:
This isn't right. Some of the changes that I made in PR #797 have disappeared here.

Also this should use 2 parameters, `$old_value` and `$value`, and do a `return` only when bailing-out, not `imagify_die()`